lib/upipe-ts/upipe_ts_decaps.c

```diff
@@ -112,7 +112,7 @@ static void upipe_ts_decaps_input(struct upipe *upipe, struct uref *uref,
                                   struct upump **upump_p)
 {
     struct upipe_ts_decaps *upipe_ts_decaps = upipe_ts_decaps_from_upipe(upipe);
-    uint8_t buffer[TS_HEADER_SIZE];
+    uint8_t buffer[TS_HEADER_SIZE + TS_HEADER_SIZE_PCR];
```

Comment:
Beware that TS_HEADER_SIZE_PCR includes TS_HEADER_SIZE. Same for TS_HEADER_SIZE_AF. Parts of the code seem wrong.